test/Autofac.Extras.Moq.Test/AutoMockFixture.cs

```diff
@@ -367,6 +367,21 @@ public void CreateNonGenericSameAsCreateGeneric()
             }
         }
 
+        [Fact]
+        public void ResolveInChildScope()
+        {
+            using (var mock = AutoMock.GetLoose())
+            {
+                mock.Mock<IDependency>().Setup(o => o.DoSomethingExternal()).Returns(1);
+                var rootScope = mock.Create<ILifetimeScope>();
```

Comment:
I'm a little concerned with folks _mocking_ `ILifetimeScope`. Mocking dependencies is one thing, but generally I'd recommend the pattern be "use real Autofac, but register mocks in the container." I raise that because I think it'd be better to show an example where the scope comes from the `AutoMock.Container` like

```c#
var rootScope = AutoMock.Container;
using var childScope = rootScope.BeginLifetimeScope();
```

Would the test still work? (I think it would.) If so, can we update the sample to show a better usage pattern?

Reply:
As I mentioned, not all these things may pertain, and it sounds like they don't. I'm not sure my position is any more "arbitrary" than your own.

It seems odd that all this came around because I didn't want the _example test_ to show mocking a lifetime scope because I want to encourage folks to avoid doing that. As noted on the test change, _it still works both ways_ so I'm super unclear what the problem really is.

However, given something about it all seems unsatisfactory, I will offer this:

If you'd like to take permanent ownership of this library, support it, and take into account as many "arbitrary viewpoints" out there as there are users and answer all those questions to their 100% satisfaction, I'd be happy to offload this particular integration. It's not something I use, there isn't anything in the way of design doc or explanation of intent, and there's no way to really know if it's doing what it's supposed to be doing. Good luck satisfying everyone.

Alternatively, if you find the library isn't doing what you want, there's not much code and it's MIT licensed. You're more than welcome to fork and make it do what you like to fit your needs. You can even fork and release a "competing" version if you like, in which case we'd probably deprecate this one to reduce overhead.

I'm sorry my answer doesn't satisfy you, but hopefully that provides you enough options to ensure you get what you need out of your testing experience.